plugins/browser-plugin-client-hints/src/index.ts

```diff
@@ -63,35 +33,63 @@ let uaClientHints: HttpClientHints;
 function forceArray<T>(array: T[] | Record<string, T>): T[] {
   if (Array.isArray(array)) return array;
 
+  // Object.keys coerces to object
+  // null and undefined throw TypeError
   return Object.keys(array).map((e) => {
     return array[e];
   });
 }
 
+/**
+ * Validates whether userAgentData is compliant to the client-hints interface.
+ * https://wicg.github.io/ua-client-hints/#interface
+ */
+function validClientHints(hints: HttpClientHints): boolean {
+  if (!hints || typeof hints.isMobile !== 'boolean' || !Array.isArray(hints.brands)) {
+    return false;
+  }
+
+  if (
+    hints.brands.some(
+      (brand) => typeof brand.brand !== 'string' || typeof brand.version !== 'string' || Object.keys(brand).length !== 2
```

Comment:
Nit: regarding the last condition that there have to be exactly 2 fields – wouldn't it be better to not take the `brands` array directly from the browser but map it and construct it on our side? This could make sure that in case the browser returns more than two properties, we can still work with the data if the attributes we need are present?

I mean something like:
```ts
const brands = hints.brands.map(b => {
   const { brand, version } = b;
   return { brand, version };
});
```

Reply:
Nice idea, i'll change so!